src/main.cpp

```diff
@@ -220,15 +220,14 @@ int main(int argc, char **argv) {
             msg.wrench.torque.z = (data[5]-8192)/calib[5];
 
             pub.publish(msg);
+            lock.unlock();
         } else {
             // Request for offset reset
             write(fdc, "O", 1);
             offset_reset_ --;
+            lock.unlock();
             cv_.notify_all();
```

Comment:
Optimization: unlock before notify_all()

See e.g. http://en.cppreference.com/w/cpp/thread/condition_variable/notify_all
_"The notifying thread does not need to hold the lock on the same mutex as the one held by the waiting thread(s); in fact doing so is a pessimization, since the notified thread would immediately block again, waiting for the notifying thread to release the lock."_